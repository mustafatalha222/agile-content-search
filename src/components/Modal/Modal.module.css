.modalOverlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.5);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 999;
}

.modalContent {
  max-width: 90%;
  width: auto;
  box-shadow: 0 4px 0 rgba(0, 0, 0, 0.3);
  position: relative;
}
