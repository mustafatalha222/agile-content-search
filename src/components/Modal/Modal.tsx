import React, { useRef, useEffect, memo } from 'react'
import styles from './Modal.module.css'

type ModalProps = {
  children: React.ReactNode
  onClose: () => void
}

const Modal = ({ children, onClose }: ModalProps) => {
  const modalRef = useRef<HTMLDivElement>(null)

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
        onClose()
      }
    }

    document.addEventListener('mousedown', handleClickOutside)
    return () => {
      document.removeEventListener('mousedown', handleClickOutside)
    }
  }, [onClose])

  return (
    <div className={styles.modalOverlay}>
      <div ref={modalRef} className={styles.modalContent}>
        {children}
      </div>
    </div>
  )
}

export default memo(Modal)
