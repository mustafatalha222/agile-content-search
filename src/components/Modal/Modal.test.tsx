import { render, screen } from '@testing-library/react'
import Modal from './Modal'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

const text = 'Test Modal'
const setup = (onClose: jest.Mock) => {
  render(<Modal onClose={onClose}>{text}</Modal>)
}

describe('Modal Component', () => {
  let onClose: jest.Mock
  beforeEach(() => {
    onClose = jest.fn()
  })

  it('renders modal content', () => {
    const onClose = jest.fn()
    setup(onClose)
    const modalContent = screen.getByText(text)
    expect(modalContent).toBeInTheDocument()
  })

  it('should not close the modal when clicking inside the modal', async () => {
    setup(onClose)
    const modalContent = screen.getByText(text)
    await userEvent.click(modalContent)
    expect(onClose).toHaveBeenCalledTimes(0)
  })

  it('should render and match snapshot', () => {
    const { asFragment } = render(<Modal onClose={onClose}>{text}</Modal>)
    expect(asFragment()).toMatchSnapshot()
  })
})
