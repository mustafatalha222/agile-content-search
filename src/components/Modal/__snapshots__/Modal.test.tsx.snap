// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalOverlay"
  >
    <div
      class="modalContent"
    >
      Test Modal
    </div>
  </div>
</DocumentFragment>
`;
